src/swap.rs

```diff
@@ -321,20 +478,55 @@ impl herc20::RedeemAsAlice
 {
     async fn redeem(
         &self,
-        params: &comit::herc20::Params,
-        deploy_event: comit::herc20::Deployed,
-    ) -> anyhow::Result<comit::herc20::Redeemed> {
+        params: &herc20::Params,
+        deploy_event: herc20::Deployed,
+    ) -> anyhow::Result<herc20::Redeemed> {
         let redeem_action = params.build_redeem_action(deploy_event.location, self.secret)?;
         let event = self.beta_wallet.redeem(redeem_action).await?;
 
         Ok(event)
     }
 }
 
-#[derive(Clone, Copy, Debug)]
+#[async_trait::async_trait]
+impl hbit::Refund for WalletAlice<BitcoinWallet, EthereumWallet, hbit::PrivateDetailsFunder> {
+    async fn refund<SC>(
+        &self,
+        params: &hbit::Params,
+        fund_event: hbit::CorrectlyFunded,
+        secp: &bitcoin::secp256k1::Secp256k1<SC>,
```

Comment:
Not sure what feels the cleanest but this is the binary application, I would not mind if secp is directly used from crate root instead of passing it around as an argument.

If we really want to pass it around as an argument then we need to remove the `pub` (and even `static` keyword from its declaration in `main.rs` and `lib.rs` and initialise it from the test/bin start.

Reply:
You are right.

I think @luckysori intention was to make this code already portable into the `comit` lib. Not sure we need to bother with that though at this stage :)

Reply:
Yep, can revert this!